gridstatus/eia.py

```diff
@@ -727,6 +754,35 @@ def _handle_fuel_type_data(df):
     return df
 
 
+def _handle_henry_hub_natural_gas_spot_prices(df):
+    # The other EIA datasets use "period" as the "Interval End" but that is not correct
+    # for this dataset because that would put the data one day ahead of how the EIA
+    # shows it here: https://www.eia.gov/dnav/ng/hist/rngwhhdD.htm
+    # We use the HENRY_HUB_TIMEZONE because the spot prices are based on delivery
+    # at Henry Hub
+    df["Interval Start"] = pd.to_datetime(df["period"]).dt.tz_localize(
+        HENRY_HUB_TIMEZONE,
+    )
```

Comment:
Updated to use US/Central time for interval start and end.

Reply:
I'm not sure this is correct because this data series also includes futures prices which are based on the NYMEX. 

> Official daily closing prices at 2:30 p.m. from the trading floor of the New York Mercantile Exchange (NYMEX) for a specific delivery month.

Should we update this series to only include the spot prices?

Reply:
Based on conversations, the answer was we should also include the futures prices (which end on April 5, 2024) and still use Central Time. We should note this in the dataset description.